edx/edx-enterprise
# ENT-215/ENT-217: Set up data transformation framework/do transformations for SAP

**Description:** 

This PR adds a framework for course metadata transformation/transmission to integrated channels.

**JIRA:** [ENT-215](https://openedx.atlassian.net/browse/ENT-215)

**Dependencies:** None

**Merge deadline:** 13 March 2017

**Installation instructions:** Requires course discovery devstack populated with course metadata

**Testing instructions:**

1. Create an `EnterpriseCustomer` with a catalog attached, and with courses present in the catalog.
2. Create an `SAPSuccessFactorsEnterpriseCustomerConfiguration` linked to the EnterpriseCustomer
3. In a devstack, run the SAP management command, and verify that the logging output shows the JSON structure that are expected for SAP. Be sure to use a username with access to the catalog in use:
`./manage.py lms transmit_courseware_data --catalog_user staff`

**Merge checklist:**

  **DO NOT** just add dependencies to `requirements/*.txt` files.

**Post merge:**
